pkg/rbd/rbd_util.go

```diff
@@ -122,25 +122,15 @@ func createImage(ctx context.Context, pOpts *rbdVolume, volSz int64, cr *util.Cr
 	image := pOpts.RbdImageName
 	volSzMiB := fmt.Sprintf("%dM", volSz)
 
-	if pOpts.ImageFormat == rbdImageFormat2 {
-		logMsg := "rbd: create %s size %s format %s (features: %s) using mon %s, pool %s "
-		if pOpts.DataPool != "" {
-			logMsg += fmt.Sprintf("data pool %s", pOpts.DataPool)
-		}
-		klog.V(4).Infof(util.Log(ctx, logMsg),
-			image, volSzMiB, pOpts.ImageFormat, pOpts.ImageFeatures, pOpts.Monitors, pOpts.Pool)
-	} else {
-		logMsg := "rbd: create %s size %s format %s using mon %s, pool %s "
-		if pOpts.DataPool != "" {
-			logMsg += fmt.Sprintf("data pool %s", pOpts.DataPool)
-		}
-		klog.V(4).Infof(util.Log(ctx, logMsg), image, volSzMiB, pOpts.ImageFormat, pOpts.Monitors, pOpts.Pool)
+	logMsg := "rbd: create %s size %s format 2 (features: %s) using mon %s, pool %s "
+	if pOpts.DataPool != "" {
+		logMsg += fmt.Sprintf("data pool %s", pOpts.DataPool)
 	}
+	klog.V(4).Infof(util.Log(ctx, logMsg),
+		image, volSzMiB, pOpts.ImageFeatures, pOpts.Monitors, pOpts.Pool)
+
+	args := []string{"create", image, "--size", volSzMiB, "--pool", pOpts.Pool, "--id", cr.ID, "-m", pOpts.Monitors, "--keyfile=" + cr.KeyFile, "--image-feature", pOpts.ImageFeatures}
```

Comment:
@dillaman if image features are empty do we need to know `--image-feature` flag?

Reply:
Yes, since otherwise it will use the system defaults where are `layering,exclusive-lock,object-map,fast-diff,deep-flatten` and krbd will fail to mount the device.

Reply:
@dillaman Thanks :+1: